bibliothecarius-service/src/main/java/com/peony/bibliothecarius/service/application/service/FileService.java

```diff
@@ -0,0 +1,78 @@
+package com.peony.bibliothecarius.service.application.service;
+
+import com.amazonaws.services.s3.AmazonS3;
+
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectInputStream;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.util.IOUtils;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class FileService {
+
+    @Value("${application.bucket.name}")
+    private String bucketName;
+
+    private  final AmazonS3 s3;
+
+    public String saveFile(MultipartFile file) {
+        String originalFilename = file.getOriginalFilename();
+        try {
+            File file1 = convertMultiPartToFile(file);
+            PutObjectResult putObjectResult = s3.putObject(bucketName, originalFilename, file1);
+            return putObjectResult.getContentMd5();
+        } catch (IOException e) {
+            throw  new RuntimeException(e);
+        }
+    }
+
+    public byte[] downloadFile(String filename) {
+        S3Object object = s3.getObject(bucketName, filename);
+        S3ObjectInputStream objectContent = object.getObjectContent();
+        try {
+            return IOUtils.toByteArray(objectContent);
+        } catch (IOException e) {
+            throw  new RuntimeException(e);
+        }
+
+
+    }
+
+    public String deleteFile(String filename) {
+
+        s3.deleteObject(bucketName,filename);
+        return "File deleted";
+    }
+
+    public List<String> listAllFiles() {
+
+        ListObjectsV2Result listObjectsV2Result = s3.listObjectsV2(bucketName);
+        return  listObjectsV2Result.getObjectSummaries().stream().map(S3ObjectSummary::getKey).collect(Collectors.toList());
```

Comment:
```suggestion
        return  listObjectsV2Result.getObjectSummaries()
                  .stream()
                  .map(S3ObjectSummary::getKey)
                  .collect(Collectors.toList());
```
when have this many chained calls better to put each on a separete line it improves readability